### PR TITLE
[Merged by Bors] - chore: Delete `Data.Nat.Units`, `Data.Int.Units`

### PR DESCRIPTION
**Pull request by YaelDillies**
The lemmas can easily move to `Algebra.Group.Nat` and `Algebra.Group.Int`, `Algebra.Ring.Int` respectively.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GCDMonoid/Nat.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/GCDMonoid/Nat.lean:8]: Order alphabetically, please

File: Mathlib/Algebra/Group/Int.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Group/Int.lean:124]:
```suggestion
```
if these really can't go here, but I'd like to understand why not

File: Mathlib/AlgebraicTopology/SimplicialSet.lean
Comment (Ruben-VandeVelde) [inline Mathlib/AlgebraicTopology/SimplicialSet.lean:10]: Order

File: Mathlib/Data/Fintype/Units.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fintype/Units.lean:8]: Order

File: Mathlib/Data/Int/AbsoluteValue.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Int/AbsoluteValue.lean:9]: You guessed it: order

File: Mathlib/Data/Ordmap/Ordset.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Ordmap/Ordset.lean:9]: Here too

**Review by Ruben-VandeVelde — COMMENTED** [review]
lgtm modulo the location of units_eq_one_or/units_ne_iff_eq_neg

File: Mathlib/Algebra/Group/Int.lean
Comment (YaelDillies) [reply Mathlib/Algebra/Group/Int.lean:124]: The `Neg ℤˣ` instance requires the `Ring ℤ` instance. In contrast, talking about elements of `ℤ` respecting `IsUnit` only requires the `CommMonoid ℤ` and `Neg ℤ` instances.

**Review by Ruben-VandeVelde — APPROVED** [review]
I see.

maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**
